addOns/retest/src/test/java/org/zaproxy/addon/retest/RetestPlanGeneratorUnitTest.java

```diff
@@ -0,0 +1,136 @@
+import org.parosproxy.paros.extension.ExtensionLoader;
+import org.parosproxy.paros.model.Model;
+import org.parosproxy.paros.network.HttpMessage;
+import org.zaproxy.addon.automation.AutomationEnvironment;
+import org.zaproxy.addon.automation.ContextWrapper;
+import org.zaproxy.addon.automation.ExtensionAutomation;
+import org.zaproxy.zap.network.HttpRequestBody;
+import org.zaproxy.zap.testutils.TestUtils;
+
+public class RetestPlanGeneratorUnitTest extends TestUtils {
+
+    private static List<AlertData> alertData = new ArrayList<>();
+
+    @BeforeAll
+    static void init() {
+        HttpMessage msgOne = new HttpMessage();
+        Alert alertOne = new Alert(100);
+        alertOne.setSource(Alert.Source.ACTIVE);
+        AlertData alertOneData = new AlertData();
+        alertOneData.setScanRuleId(100);
+        alertOneData.setAlertName("Test Alert One");
+        alertOneData.setUrl("https://www.exampleone.com");
+        alertOneData.setMethod("GET");
+        alertOneData.setAttack("Test Attack One");
+        alertOneData.setParam("Test Param One");
+        alertOneData.setEvidence("Test Evidence One");
+        alertOneData.setConfidence("Test Confidence One");
+        alertOneData.setRisk("Test Risk One");
+        alertOneData.setOtherInfo("Test Other Info One");
+        alertOneData.setMsg(msgOne);
+        alertOneData.setAlert(alertOne);
+
+        HttpMessage msgTwo = new HttpMessage();
+        msgTwo.setRequestBody(new HttpRequestBody("Test Body"));
+        Alert alertTwo = new Alert(200);
+        alertTwo.setSource(Alert.Source.PASSIVE);
+        AlertData alertTwoData = new AlertData();
+        alertTwoData.setScanRuleId(200);
+        alertTwoData.setAlertName("Test Alert Two");
+        alertTwoData.setUrl("https://www.exampletwo.com");
+        alertTwoData.setMethod("POST");
+        alertTwoData.setAttack("Test Attack Two");
+        alertTwoData.setParam("Test Param Two");
+        alertTwoData.setEvidence("Test Evidence Two");
+        alertTwoData.setConfidence("Test Confidence Two");
+        alertTwoData.setRisk("Test Risk Two");
+        alertTwoData.setOtherInfo("Test Other Info Two");
+        alertTwoData.setMsg(msgTwo);
+        alertTwoData.setAlert(alertTwo);
+
+        alertData.add(alertOneData);
+        alertData.add(alertTwoData);
+    }
+
+    @Override
+    protected void setUpMessages() {
+        mockMessages(new ExtensionAutomation());
+        mockMessages(new ExtensionRetest());
+    }
+
+    @BeforeEach
+    void setUp() throws Exception {
+        ExtensionLoader extensionLoader = mock(ExtensionLoader.class);
+        Control.initSingletonForTesting(mock(Model.class), extensionLoader);
+    }
+
+    @Test
+    void shouldGeneratePlan() {}
+
+    @Test
+    void shouldGenerateEnvironment() {
+        // Given
+        RetestPlanGenerator retestGen = new RetestPlanGenerator(alertData);
+
+        // When
+        AutomationEnvironment env = retestGen.getPlan().getEnv();
```

Comment:
The NPE caused by getPlan() can be addressed by adding `Constant.messages = new I18N(Locale.ENGLISH);` in the `//Given` block. Subsequently there are problems with the i18n of enums, I think that can likely be addressed with further mocking and when/thenReturn handling. I'd suggest looking at the automation UnitTests there's extensive using of mock/when/thenReturn in those tests which likely apply here as well.

Reply:
@thc202 suggested handling the i18n problems with the enums by also calling the `setUpZap` method in addition to `mockMessages`, and that worked. As for the `Constant.messages` line, I added it, but that doesnt seem to work.

It seems to me that you can only load messages from one extension for a test.

Reply:
Okay